File: plugin/shared/src/main/java/com/jonathanpotts/cartographytable/shared/models/VectorXYZ.java
package com.jonathanpotts.cartographytable.shared.models;

/**
 * Stores a 3D vector containing a X, Y, and Z coordinate.
 */
public class VectorXYZ {
  /**
   * X coordinate.
   */
  public int x;

  /**
   * Y coordinate.
   */
  public int y;

  /**
   * Z coordinate.
   */
  public int z;

  /**
   * Creates a 3D vector containing a X, Y, and Z coordinate.
   *
   * @param x X coordinate.
   * @param y Y coordinate.
   * @param z Z coordinate.
   */
  public VectorXYZ(int x, int y, int z) {
    this.x = x;
    this.y = y;
    this.z = z;
  }
}

File: src/main/kotlin/io/github/jonathanpotts/cartographytable/models/VectorXYZ.kt
package io.github.jonathanpotts.cartographytable.models

import kotlinx.serialization.Serializable

/**
 * Stores a 3D vector containing a X, Y, and Z coordinate.
 *
 * @property x X coordinate.
 * @property y Y coordinate.
 * @property z Z coordinate.
 */
@Serializable
data class VectorXYZ(
    var x: Int = 0,
    var y: Int = 0,
    var z: Int = 0
)
